packages/leafygreen-provider/src/UsingKeyboardProvider.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { createContext, useContext, useState, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { useEventListener } from '@leafygreen-ui/hooks';
+
+interface UsingKeyboardState {
+  usingKeyboard?: boolean;
+  setUsingKeyboard: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const UsingKeyboardContext = createContext<UsingKeyboardState>({
+  usingKeyboard: undefined,
+  setUsingKeyboard: () => {},
+});
+
+// All keys here are used to manage focus through keyboard interaction.
+export const NavigationKeyCodes: { readonly [k: string]: number } = {
+  tab: 9,
+  leftArrow: 37,
+  upArrow: 38,
+  rightArrow: 39,
+  downArrow: 40,
+} as const;
+
+export function useShowFocus(): boolean {
+  const { usingKeyboard } = useContext(UsingKeyboardContext);
+
+  if (usingKeyboard == null) {
+    // If there's no context provider available, we show focus states as usual.
+    return true;
+  }
+
+  return usingKeyboard;
+}
+
+export function useUsingKeyboardContext(): UsingKeyboardState {
+  return useContext(UsingKeyboardContext);
+}
+
+interface UsingKeyboardProviderProps {
+  children?: React.ReactNode;
+}
+
+function UsingKeyboardProvider({ children }: UsingKeyboardProviderProps) {
+  const [usingKeyboard, setUsingKeyboard] = useState(false);
+
+  useEventListener('mousedown', () => setUsingKeyboard(false), {
```

Comment:
I suppose we don't need to enable / disable them, but I'm not sure what case you see where it's more expensive to register them when needed rather than having them be persistent. They should only swap on user interaction, so the actual registering of the events should have a negligible effect on performance.

That said, I'm not convinced it will have a large impact either way, and am happy to leave both event listeners always enabled if you strongly feel this will be a performance bottleneck.

Reply:
Meh, let's keep it as is and see how it goes. I have no proof either way and not really concerned to measure it now.